spec.bs

```diff
@@ -2741,6 +2777,59 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   [=automatic beacon data/once=] set to true.
 </div>
 
+<div algorithm=get-initiator-ancestor-root-origin>
+  To <dfn>get the initiator ancestor root origin</dfn> of a given {{Document}} |sourceDocument|:
+
+  1. Let |navigable| be |sourceDocument|'s [=node navigable=].
+
+  1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
```

Comment:
When would `|navigable|` be null? First, if we call this on a detached document I guess, but navigations cannot be initiated there so I don't think that would cause it. Second... if we called this algorithm on a document that has nothing to do with a fenced frame tree, then I guess we'd just reach the top-level traversable's parent (null) and that's when we'd get kicked out of this loop with a null navigable, right?

And that's only possible because we invoke this algorithm indiscriminately for all navigations? Do I have that right?

Reply:
> if we called this algorithm on a document that has nothing to do with a fenced frame tree, then I guess we'd just reach the top-level traversable's parent (null) and that's when we'd get kicked out of this loop with a null navigable

> that's only possible because we invoke this algorithm indiscriminately for all navigations

Yes, that's why this is in place. This will also happen for fenced frames that don't have a fenced frame ancestor, since it will cross the fenced boundary, then traverse up the non-fenced tree until it reaches the root.